sickbeard/helpers.py

```diff
@@ -1081,7 +1081,9 @@ def backupConfigZip(fileList, archive, arcname = None):
 
 def restoreConfigZip(archive, targetDir):
     try:
-        if not os.path.exists(targetDir):
+        if os.path.exists(targetDir):
+            shutil.move(targetDir,os.path.join(targetDir, '{0}.bak-{1}'))        

```

Comment:
This wont work as you intended. You need to extract the leaf from the targetdir, then append the string you want(bak-date) then append it to the base of targetDir.